Syslog network backend

jukkar opened the pull request:
This is a network backend for syslog. It implements https://tools.ietf.org/html/rfc5424 and https://tools.ietf.org/html/rfc5426 in order to send syslog messages to remote syslog server like syslog-ng.

jukkar committed:
syslog: Add networking backend

If network based syslog backend is enabled in Kconfig file,
then syslog messages are sent to external system using UDP.
See RFC 5424 and RFC 5426 for details about the syslog protocol.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
net: app: Enable syslog to network in init

Start to send syslog messages to net after we have a proper
IP address.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
samples: net: syslog: Remote syslog service example

The sample application will send syslog messages to remote
syslog server.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jhedberg commented on subsys/logging/sys_log_net.c:
```diff
@@ -0,0 +1,224 @@
+
+	va_end(vargs);
+
+	len = strlen(ptr);
```
Isn't this redundant? You already have the string length stored in the ret variable.

jukkar replied:
Yes, I realized it too after sending the PR. Will fix this.

jhedberg commented on subsys/logging/sys_log_net.c:
```diff
@@ -0,0 +1,224 @@
+ * function, use some pre-defined values.
+ */
+static const int facility = 16; /* local0 */
+static const int severity = 6; /* info */
```
Note that with the planned LOG_* API (which will replace SYS_LOG) you won't need to do this kind of hard-coding, since you'll be getting the logging level as proper meta-data (with SYS_LOG it's just embedded as a prefix in the string)

jukkar replied:
Good that this issue with needed func parameters in being planned in the future.

dbkinder commented on samples/net/syslog_net/README.rst:
```diff
@@ -0,0 +1,33 @@
+Overview
+********
+
+This sample application enables remote syslog service. You need to
```
A bit clearer would be:

    This sample application enables a remote syslog service that will
    send syslog messages to a remote server, as configured in ``prj.conf``.

dbkinder commented on samples/net/syslog_net/README.rst:
```diff
@@ -0,0 +1,33 @@
+************
+
+- :ref:`networking_with_qemu`
+
```
Looks like you could use some information here about the options you need to configure in ``prj.conf`` to specifically make remote syslog work.  Is it just the lines in ``prj.conf`` after the comment ``# syslog net backend config`` ?

jukkar commented:
Uploaded new version according to comments.

jukkar committed:
net: app: Fix debug level check

We need to check the debug level to be INFO or higher, otherwise
the some of the variables in net app init.c will give unused
warning. Also the sys_log.h needs to be included _after_ we have
set the logging level, this is done by net_core.h so we should
not include the syslog header file here.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

dbkinder commented on samples/net/syslog_net/README.rst:
```diff
@@ -0,0 +1,52 @@
+   :goals: build
+   :compact:
+
+For configuring the remote IPv6 syslog server, set the following
```
Do you need to make these changes to prj.conf before doing the build step?  If so, this should move before the build step above.

jukkar commented:
Documentation change and sanitycheck fix.

dbkinder reviewed (approved):
thanks

pfalcon commented:
At least this defines separate pkt/net_buf pool for syslog, so no concerns how this affects of the main IP stack buffers.

jukkar commented:
> At least this defines separate pkt/net_buf pool for syslog, so no concerns how this affects of the main IP stack buffers.

Yes, using separate pool for this stuff is really mandatory so that this feature does not affect general IP stack functionality.
